src/Event/ErrorEvent.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Error event to be used by the payment gateway logger.
+ *
+ * @package    payment-gateway-logger
+ * @author     manzoj
+ * @version    1
+ */
+
+namespace PaymentGatewayLogger\Event;
+
+use Exception;
+use Guzzle\Common\Event;
+
+class ErrorEvent extends Event
+{
+    /**
+     * @var Exception
+     */
+    protected $error;
+
+    /**
+     * @var string
+     */
+    protected $type = Constants::OMNIPAY_REQUEST_ERROR;
```

Comment:
any reason that this instance variable is needed instead of just making getType() return the const directly?

Reply:
Ehh don't remember why I want this way tbh :) I actually meant to do what you suggested above ^